// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/conformance/expressions/superPropertyAccess/superSymbolIndexedAccess1.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -2,17 +2,19 @@
 
 //// [superSymbolIndexedAccess1.d.ts]
 declare var symbol: invalid;
 declare class Foo {
+    [symbol](): invalid;
 }
 declare class Bar extends Foo {
+    [symbol](): invalid;
 }
 
 /// [Errors] ////
 
 superSymbolIndexedAccess1.ts(1,14): error TS9010: Variable must have an explicit type annotation with --isolatedDeclarations
-superSymbolIndexedAccess1.ts(4,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
-superSymbolIndexedAccess1.ts(10,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+superSymbolIndexedAccess1.ts(4,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
+superSymbolIndexedAccess1.ts(10,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
 
 
 ==== superSymbolIndexedAccess1.ts (3 errors) ====
     var symbol = Symbol.for('myThing');
@@ -22,16 +24,18 @@
     
     class Foo {
         [symbol]() {
         ~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
+!!! related TS9034 superSymbolIndexedAccess1.ts:4:5: Add a return type to the method
             return 0;
         }
     }
     
     class Bar extends Foo {
         [symbol]() {
         ~~~~~~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
+!!! related TS9034 superSymbolIndexedAccess1.ts:10:5: Add a return type to the method
             return super[symbol]();
         }
     }
\ No newline at end of file
